File: src/api/i18n.ts
/**
 * 国际化 API 封装
 * i18n API wrapper
 *
 * 封装语言获取、翻译加载和语言切换。
 * Wraps locale retrieval, translation loading, and language switching.
 */
import { invoke } from '@tauri-apps/api/core'

/** 翻译字典类型 / Translation dictionary type */
export type Translations = Record<string, string>

/**
 * 函数名称：getLocale
 * 函数作用：获取当前应用语言环境。
 * Purpose: Gets the current application locale.
 * 调用方：i18nStore 初始化
 * Called by: i18nStore initialization
 * 中文关键词：获取语言，语言环境，locale
 * English keywords: get locale, language environment
 */
export async function getLocale(): Promise<string> {
  return await invoke('get_locale')
}

/**
 * 函数名称：getTranslations
 * 函数作用：加载指定语言的翻译文件。
 * Purpose: Loads translation file for the specified locale.
 * 参数 locale: 语言代码 (zh-CN / en-US)
 * Called by: i18nStore 初始化
 * 中文关键词：加载翻译，语言包，翻译文件
 * English keywords: load translations, language pack, translation file
 */
export async function getTranslations(locale: string): Promise<Translations> {
  return await invoke('get_translations', { locale })
}

/**
 * 函数名称：setLocale
 * 函数作用：设置当前应用语言环境并持久化。
 * Purpose: Sets the current locale and persists it.
 * 参数 locale: 语言代码
 * 副作用：写入 config/app.json
 * Called by: SettingsPage 语言选择器
 * 中文关键词：设置语言，切换语言
 * English keywords: set locale, switch language
 */
export async function setLocale(locale: string): Promise<boolean> {
  return await invoke('set_locale', { locale })
}

/**
 * 简易翻译函数 — 从翻译字典查找文本
 * Simple translation function — looks up text from translation dictionary
 */
export function translate(key: string, translations: Translations, fallback?: string): string {
  return translations[key] || fallback || key
}

File: src-tauri/src/commands/i18n.rs
// 国际化命令 — 语言切换和翻译加载
// i18n commands — language switching and translation loading
use std::sync::{Arc, Mutex};

/// 函数名称：resolve_i18n_path
/// 函数作用：解析 i18n 资源路径，优先 CWD，其次上级目录（兼容 tauri dev）。
/// Purpose: Resolves i18n resource path, preferring CWD then parent dir (for tauri dev).
fn resolve_i18n_path(locale: &str) -> std::path::PathBuf {
    let local = std::path::PathBuf::from(format!("config/i18n/{}.json", locale));
    if local.exists() {
        return local;
    }
    let parent = std::path::PathBuf::from(format!("../config/i18n/{}.json", locale));
    if parent.exists() {
        return parent;
    }
    local
}

/// 函数名称：get_locale
/// 函数作用：从内存中的 AppConfig 状态获取当前语言环境。
/// Purpose: Gets the current application locale from in-memory AppConfig state.
/// Returns: 语言代码 (zh-CN / en-US) / Language code (zh-CN / en-US)
/// 调用方：前端初始化
/// Called by: Frontend initialization
/// 中文关键词：语言设置，国际化，获取语言
/// English keywords: language settings, i18n, get locale
#[tauri::command]
pub async fn get_locale(
    config: tauri::State<'_, Arc<Mutex<crate::models::config::AppConfig>>>,
) -> Result<String, String> {
    let cfg = config.lock().map_err(|e| e.to_string())?;
    Ok(cfg.locale.clone())
}

/// 函数名称：get_translations
/// 函数作用：加载指定语言的翻译文件。
/// Purpose: Loads translations for the specified locale.
/// 参数 locale: 语言代码 (zh-CN, en-US) / Language code
/// Returns: 翻译 JSON 对象 / Translation JSON object
/// 调用方：前端 i18nStore 初始化
/// Called by: Frontend i18nStore initialization
/// 副作用：从 config/i18n/{locale}.json 读取文件
/// Side effect: Reads from config/i18n/{locale}.json
/// 中文关键词：翻译加载，语言包，国际化文本
/// English keywords: translation loading, language pack, i18n text
#[tauri::command]
pub async fn get_translations(locale: String) -> Result<serde_json::Value, String> {
    let path = resolve_i18n_path(&locale);
    let content = std::fs::read_to_string(&path)
        .map_err(|e| format!("加载语言文件失败 ({}): {}", path.display(), e))?;
    let translations: serde_json::Value =
        serde_json::from_str(&content).map_err(|e| format!("解析语言文件失败: {}", e))?;
    Ok(translations)
}

/// 函数名称：set_locale
/// 函数作用：设置当前应用语言环境，更新内存状态并持久化到配置文件。
/// Purpose: Sets the current application locale, updates in-memory state and persists to config file.
/// 参数 locale: 语言代码 / Language code
/// 副作用：更新 AppConfig 内存状态，写入 config/app.json
/// Side effect: Updates AppConfig in-memory state and writes to config/app.json
/// 调用方：前端 SettingsPage 语言选择器
/// Called by: Frontend SettingsPage language selector
/// 中文关键词：设置语言，切换语言，语言选择
/// English keywords: set locale, switch language, language selection
#[tauri::command]
pub async fn set_locale(
    locale: String,
    config: tauri::State<'_, Arc<Mutex<crate::models::config::AppConfig>>>,
) -> Result<bool, String> {
    // 校验 locale 值，防止路径穿越 / Validate locale to prevent path traversal
    if !locale.chars().all(|c| c.is_ascii_alphanumeric() || c == '-' || c == '_') {
        return Err("Invalid locale format".to_string());
    }
    // 验证语言文件存在 / Verify language file exists
    let i18n_path = resolve_i18n_path(&locale);
    if !i18n_path.exists() {
        return Err(format!("Language file not found for locale: {}", locale));
    }
    // 更新内存中的 AppConfig / Update in-memory AppConfig
    {
        let mut cfg = config.lock().map_err(|e| e.to_string())?;
        cfg.locale = locale.clone();
    }
    // 通过 AppConfig::save() 持久化 / Persist via AppConfig::save()
    {
        let cfg = config.lock().map_err(|e| e.to_string())?;
        cfg.save().map_err(|e| format!("保存配置失败: {}", e))?;
    }
    Ok(true)
}
